bindings/rust/s2n-tls-tokio/src/contract.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::{
+    sync::atomic::{AtomicBool, Ordering},
+    task::{Context, Poll, Waker},
+};
+use std::{sync::Arc, task::Wake};
+
+/// Checks that if a function returns [`Poll::Pending`], then the function called [`Waker::clone`],
+/// [`Waker::wake`], or [`Waker::wake_by_ref`] on the [`Context`]'s [`Waker`].
+pub struct Contract {
```

Comment:
Idk if this was clearer from the module / context in s2n-quic, but isn't "Contract" pretty general here? Just reading s2n-tls-tokio code, I don't think it'd be obvious to me that this was related to wakers. There are a lot of possible "contract"s we could be asserting.

Reply:
I can put it in the same nested path as s2n-quic.